dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -447,8 +453,23 @@ default String getURLFromEntry(final String dockerEntry, final ToolDAO toolDAO,
                 // when we found a published tool, link to the tool on Dockstore
                 url = dockstorePath + dockerImage;
             }
+        } else if (registry.isPresent() && registry.get().equals(Registry.AMAZON_ECR)) {
+            List<Tool> publishedByPath = toolDAO.findAllByPath(dockerImage, true);
```

Comment:
I think you could just use `toolDAO.findByPath()`.

Reply:
Took a look and `toolDAO.findByPath()` looks for a tool name in the path, which isn't really applicable here. `toolDAO.findAllByPath()` ignores tool name